qiskit/aqua/operators/gradients/circuit_qfis/overlap_block_diag.py

```diff
@@ -0,0 +1,305 @@
+
+        generators = self._get_generators(params, circuit)
+
+        blocks = []
+
+        # Psi_i = layer_i @ layer_i-1 @ ... @ layer_0 @ Zero
+        for k, psi_i in enumerate(psis):
+            params = block_params[k]
+            block = np.zeros((len(params), len(params))).tolist()
+
+            # calculate all single-operator terms <psi_i|generator_i|psi_i>
+            single_terms = np.zeros(len(params)).tolist()
+            for i, p_i in enumerate(params):
+                generator = generators[p_i]
+                psi_gen_i = ~StateFn(generator) @ psi_i @ Zero
+                psi_gen_i = PauliExpectation().convert(psi_gen_i)
+                single_terms[i] = psi_gen_i
+
+            def get_parameter_expression(circuit, param):
+                if len(circuit._parameter_table[param]) > 1:
+                    raise NotImplementedError("OverlapDiag does not yet support multiple "
+                                              "gates parameterized by a single parameter. For such "
+                                              "circuits use LinCombFull")
+                gate = circuit._parameter_table[param][0][0]
+                if len(gate.params) > 1:
+                    raise AquaError("OverlapDiag cannot yet support gates with more than one "
+                                    "parameter.")
+
+                param_value = gate.params[0]
+                return param_value
+
+            # Calculate all double-operator terms <psi_i|generator_j @ generator_i|psi_i>
+            # and build composite operators for each matrix entry
+            for i, p_i in enumerate(params):
+                generator_i = generators[p_i]
+                param_expr_i = get_parameter_expression(circuit, p_i)
+
+                for j, p_j in enumerate(params):
+                    if i == j:
+                        block[i][i] = ListOp([single_terms[i]], combo_fn=lambda x: 1 - x[0] ** 2)
+                        if isinstance(param_expr_i, ParameterExpression) and not isinstance(
+                                param_expr_i, Parameter):
+                            expr_grad_i = DerivativeBase.parameter_expression_grad(
+                                param_expr_i, p_i)
+                            block[i][j] *= (expr_grad_i) * (expr_grad_i)
+                        continue
+
+                    generator_j = generators[p_j]
+                    generator = ~generator_j @ generator_i
+                    param_expr_j = get_parameter_expression(circuit, p_j)
+
+                    psi_gen_ij = ~StateFn(generator) @ psi_i @ Zero
+                    psi_gen_ij = PauliExpectation().convert(psi_gen_ij)
+                    cross_term = ListOp([single_terms[i], single_terms[j]], combo_fn=np.prod)
+                    block[i][j] = psi_gen_ij - cross_term
+
+                    if isinstance(param_expr_i, ParameterExpression) and not isinstance(
+                            param_expr_i, Parameter):
```

Comment:
```suggestion
                    # pylint: disable=unidiomatic-typecheck
                    if type(param_expr_i) == ParameterExpression: 
```

Reply:
Yes this is better

Reply:
But now the check is missing whether it is a Parameter bc in that case you do not need to execute the 'parameter_expression_grad'.

Reply:
As a 'Parameter' is also of type 'ParameterExpression'.

Reply:
The `==` type check checks for exactly that type. It is not true for subclasses 😄

Reply:
I learned so much today :)